packages/core/src/scripts/migrate-legacy-project.ts

```diff
@@ -0,0 +1,243 @@
+  }
+
+  return {
+    migrations,
+    migrationData,
+  };
+}
+
+async function findManifests(): Promise<string[]> {
+  const files = await fs.readdir(OPEN_ZEPPELIN_FOLDER);
+  return files.filter(isManifestFile).map(preppendPath);
+}
+
+function preppendPath(location: string): string {
+  return path.join(OPEN_ZEPPELIN_FOLDER, location);
+}
+
+function isManifestFile(fileName: string): boolean {
+  const network = getNetworkName(fileName);
+  return isTestnet(network) || isDevnet(network);
+}
+
+function getNetworkName(fileName: string): string {
+  return path.basename(fileName, '.json');
+}
+
+function isDevnet(network: string) {
+  return network.includes('dev');
+}
+
+function isTestnet(network: string) {
+  return ['mainnet', 'rinkeby', 'ropsten', 'kovan', 'goerli'].includes(network);
+}
+
+function getNewManifestLocation(oldName: string): string {
+  return isDevnet(oldName) ? oldName.replace('dev', 'unknown') : oldName;
+}
+
+async function writeJSONFile(location: string, data: unknown): Promise<void> {
+  await fs.writeFile(location, JSON.stringify(data, null, 2));
+}
+
+function transformProxies(proxies: LegacyProxies) {
+  return Object.keys(proxies).map(proxyName => ({
+    [path.basename(proxyName)]: proxies[proxyName].map(proxy => ({
+      address: proxy.address,
+      implementation: proxy.implementation,
+      admin: proxy.admin,
+    })),
+  }));
+}
+
+function updateManifest(oldManifest: LegacyManifest): ManifestData {
+  const proxyAdmin = oldManifest.proxyAdmin.address;
+
+  if (proxyAdmin === undefined) {
+    throw new Error('Legacy manifest does not have admin address');
+  }
+
+  if (Object.keys(oldManifest.solidityLibs).length === 0) {
+    throw new Error('Legacy manifest uses external Solidity libraries');
+  }
+
+  return {
+    manifestVersion: '3.1',
+    impls: transformImplementations(oldManifest.contracts),
+    admin: {
+      address: proxyAdmin,
+    },
+  };
+}
+
+function transformImplementations(contracts: LegacyContracts): Record<string, ImplDeployment> {
+  const impls: Record<string, ImplDeployment> = {};
+  for (const contractName in contracts) {
+    const contract = contracts[contractName];
+    if (contract.deployedBytecodeHash === undefined) {
+      continue;
+    } else {
+      impls[contract.deployedBytecodeHash] = transformImplementationItem(contract);
+    }
+  }
+  return impls;
+}
+
+function transformImplementationItem(contract: LegacyContract): ImplDeployment {
+  return {
+    address: contract.address || 'NO_ADDRESS_FOUND',
+    layout: transformLayout(contract),
+  };
+}
+
+function transformLayout(contract: LegacyContract): StorageLayout {
+  return {
+    storage: contract.storage?.map(transformStorageItem) || [],
+    types: transformTypes(contract.types || {}),
+  };
+}
+
+function transformStorageItem(storageItem: LegacyStorageItem): StorageItem {
+  return {
+    contract: storageItem.contract,
+    label: storageItem.label,
+    type: storageItem.type,
```

Comment:
Have you looked into the differences between the CLI `typeIdentifier` processing and ours? If not this is just a reminder. I am pretty sure we do very different stuff there and I think it's worth stopping everything and taking a good look at how we can handle that.

Reply:
I did and it does different stuff, this is another issue yet to be solved.